app.rb

```diff
@@ -0,0 +1,174 @@
+require_relative 'person'
+require_relative 'classroom'
+require_relative 'book'
+require_relative 'rental'
+require_relative 'student'
+require_relative 'teacher'
+
+class App
+  attr_reader :people, :books, :rentals
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_books
+    puts 'Listing all books:'
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    puts 'Listing all people:'
+    @people.each do |person|
+      person_type = person.is_a?(Student) ? 'Student' : 'Teacher'
+      puts "[#{person_type}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  def list_people_with_numbers
+    puts 'Listing all people:'
+    @people.each_with_index do |person, index|
+      person_type = person.is_a?(Student) ? 'Student' : 'Teacher'
+      puts "#{index + 1}. #{person_type}: #{person.name}"
+    end
+  end
+
+  def create_person(type, name = nil, age = nil, parent_permission = nil)
+    case type
+    when 'student'
+      name = prompt_name(name)
+      age = prompt_age(age)
+      parent_permission = prompt_parent_permission(parent_permission)
+
+      @people << Student.new(age, name: name, parent_permission: parent_permission)
+    when 'teacher'
+      age = prompt_age(age)
+      name = prompt_name(name)
+      specialization = prompt_specialization
+
+      teacher = Teacher.new(age, specialization, name: name)
```

Comment:
- [ ] As I mentioned earlier, the same bug was found here. so please consider changing this line to `Teacher.new(age, specialization, name)` to fix the syntax error.